internal/pkg/agent/cmd/run.go

```diff
@@ -287,6 +295,27 @@ func runElasticAgent(ctx context.Context, cancel context.CancelFunc, override cf
 	}
 	defer control.Stop()
 
+	// create symlink from /run/elastic-agent.sock to `paths.ControlSocket()` when running as root
+	// this provides backwards compatibility as the control socket was moved with the addition of --unprivileged
+	// option during installation
+	if isRoot && paths.RunningInstalled() && paths.ControlSocketRunSymlink != "" {
+		socketPath := strings.TrimPrefix(paths.ControlSocket(), "unix://")
+		socketLog := controlLog.With("path", socketPath).With("link", paths.ControlSocketRunSymlink)
+		// ensure it doesn't exist before creating the symlink
+		if err := os.Remove(paths.ControlSocketRunSymlink); err != nil && !errors.Is(err, os.ErrNotExist) {
+			socketLog.Errorf("Failed to remove existing control socket symlink %s: %s", paths.ControlSocketRunSymlink, err)
+		}
+		if err := os.Symlink(socketPath, paths.ControlSocketRunSymlink); err != nil {
+			socketLog.Errorf("Failed to create control socket symlink %s -> %s: %s", paths.ControlSocketRunSymlink, socketPath, err)
+		} else {
+			socketLog.Infof("Created control socket symlink %s -> %s; allowing unix://%s connection", paths.ControlSocketRunSymlink, socketPath, paths.ControlSocketRunSymlink)
+		}
+		defer func() {
+			// delete the symlink on exit; ignore the error
+			_ = os.Remove(paths.ControlSocketRunSymlink)
```

Comment:
Would it be valuable to log the error even if it is still ignored?

Reply:
I agree it'd be good to log it

Reply:
Added log.